### ISSUE TITLE
Video samples exceeding bucket resolutions are not downscaled during training, causing OOM，wan2.2 lora trian，

### ISSUE THREAD

**Issue by wzgrx** [kohya-ss/musubi-tuner#619]
[general]
caption_extension = ".txt"
batch_size = 1
enable_bucket = false

[[datasets]]
image_directory = "E:/AI/musubi-tuner/train/image"
cache_directory = "E:/AI/musubi-tuner/train/cache_image"
resolution = [1024, 1024]   
num_repeats = 1

[[datasets]]
video_directory = "E:/AI/musubi-tuner/train/video"
cache_directory = "E:/AI/musubi-tuner/train/cache_video"
frame_extraction = "full"
source_fps = 16.0
target_frames = [41]
max_frames = 41
enable_bucket = true
bucket_no_upscale = false
resolution = [768, 768]

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision fp16 `
  .\src\musubi_tuner\wan_train_network.py `
  --task i2v-A14B `
  --dataset_config "E:\AI\musubi-tuner\train\2.toml" `
  --dit "E:\AI\musubi-tuner\wan\wan2.2_i2v_high_noise_14B_fp16.safetensors" `
  --xformers `
  --fp8_base --fp8_scaled `
  --optimizer_type adamw8bit --learning_rate 2e-4 `
  --gradient_checkpointing `
  --max_data_loader_n_workers 8 --persistent_data_loader_workers `
  --network_module networks.lora_wan --network_dim 16 --network_alpha 16 `
  --timestep_sampling shift --discrete_flow_shift 5.0 `
  --min_timestep 900 --max_timestep 1000 `
  --preserve_distribution_shape `
  --force_v2_1_time_embedding `
  --max_train_epochs 10 --save_every_n_epochs 1 --seed 42 `
  --output_dir "E:\AI\musubi-tuner\out" --output_name "my_i2v_lora_high" `
  --blocks_to_swap 16 `
  --offload_inactive_dit

When training with mixed image and video datasets using enable_bucket = true, video files with resolution higher than any defined bucket_resolutions (e.g., 1024x1024 videos when max bucket is [768,768]) are NOT automatically downscaled to the nearest valid bucket. Instead, they are processed at their original resolution, leading to unexpected high VRAM usage and CUDA out-of-memory (OOM) errors.

This behavior contradicts the expected bucketing logic, where all samples—images and videos—should be resized to the closest supported bucket resolution to maintain consistent batch memory footprint.

🔸 Note: Image samples appear to respect bucket resizing correctly. The issue seems specific to video frames.

🔄 Steps to Reproduce
Prepare a dataset containing:
Images at 1024x1024
Videos at 1024x1024 or higher (e.g., 1280x720)
Configure dataset_config.toml with:
TOML
enable_bucket = true
bucket_resolutions = [[512,512], [512,768], [768,512], [768,768]]
# Note: NO [1024,1024] bucket defined
Run training with --preserve_distribution_shape (or without).
Observe that:
Video frames are not resized to 768x768
Model attempts to process 1024x1024 video latents
Training crashes with OOM on GPUs with ≤24GB VRAM
✅ Expected Behavior
All input samples (images and videos) should be resized to the nearest bucket resolution defined in bucket_resolutions.
If no bucket matches, the sample should either:
Be downscaled to the largest available bucket (e.g., 768x768), or
Be skipped with a warning (if bucket_no_upscale = true and upscale is required)
This ensures predictable VRAM usage and alignment with standard bucketing practices in diffusion training.

❌ Actual Behavior
High-resolution videos bypass bucket resizing logic.
They are encoded and trained at native resolution.
Causes uncontrolled VRAM spikes and training instability.
🖥️ Environment
Musubi-Tuner version: [e.g., v0.9.5 or git commit hash]
Base model: wan2.2_i2v_high_noise_14B_fp16.safetensors
GPU: [e.g., RTX 4090 24GB]
OS: Windows 11 / Linux
Command used:
BASH
python train.py --dataset_config dataset.toml --preserve_distribution_shape ...
💡 Suggested Fix
Ensure video frame preprocessing applies the same bucket resolution matching and resizing logic as used for static images. Specifically:

After frame extraction, compute aspect ratio and select nearest bucket.
Resize frames before latent caching or training.
Respect bucket_no_upscale flag for videos as well.
